tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -850,5 +850,523 @@ int main(int argc, char **argv)
+                    /* Not valid for connection (does not meet minimum version requirement and higher_vers_match already found) */
+                    &s2n_dhe_rsa_with_aes_256_gcm_sha384
+                };
+
+                cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(test_cipher_suite_list),
+                    .suites = test_cipher_suite_list,
+                    .allow_chacha20_boosting = true,
+                };
+
+                uint8_t chacha20_boosted_wire[] = {
+                    0x00,
+                    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
+                };
+                uint8_t count = sizeof(chacha20_boosted_wire) / S2N_SSLv2_CIPHER_SUITE_LEN;
+
+                /* Verify server negotiate its higher_vers_match ecdhe_ecdsa_aes_128_cbc instead of a chacha20 ciphersuite */
+                cipher_preferences.allow_chacha20_boosting = true;
+                EXPECT_SUCCESS(s2n_set_cipher_as_sslv2_server(connection, chacha20_boosted_wire, count));
+                EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha);
+            }
+
+            /* Server is able to negotiate a ciphersuite even without chacha20 in its cipher pref */
+            {
+                DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+                EXPECT_NOT_NULL(connection);
+                EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+
+                connection->security_policy_override = &security_policy;
+                connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(connection, S2N_TLS13));
+
+                static struct s2n_cipher_suite *test_cipher_suite_list[] = {
+                    /* Negotiated because it is the server's most preferred */
+                    &s2n_tls13_aes_128_gcm_sha256,                  
+                    /* Not considered */
+                    &s2n_tls13_aes_256_gcm_sha384,                  
+                };
+
+                cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(test_cipher_suite_list),
+                    .suites = test_cipher_suite_list,
+                    .allow_chacha20_boosting = true,
+                };
+
+                /*
+                 * Client did not signal chacha20 boosting.
+                 * TLS_AES_128_GCM_SHA256 > TLS_AES_256_GCM_SHA384 
+                 */
+                uint8_t wire_without_chacha20[] = {
+                    TLS_AES_256_GCM_SHA384,
+                    TLS_AES_128_GCM_SHA256,
+                };
```

Comment:
Doesn't the client need to trigger chacha20 boosting for this to be an interesting test case? Otherwise, it's just "Chacha20 boosting is disabled when client did not indicate chacha20 preference".